src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+namespace Acquia\Cli\Command\Push;
+
+use Acquia\Cli\Command\Pull\PullCommandBase;
+use Acquia\Cli\Exception\AcquiaCliException;
+use Acquia\Cli\Output\Checklist;
+use Closure;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\Finder;
+use Webmozart\PathUtil\Path;
+
+/**
+ * Class PushArtifactCommand.
+ */
+class PushArtifactCommand extends PullCommandBase {
+
+  protected static $defaultName = 'push:artifact';
+
+  /**
+   * @var string
+   *
+   * Drupal project directory.
+   */
+  protected $dir;
+
+  /**
+   * @var array
+   *
+   * Composer vendor directories.
+   */
+  protected $vendorDirs;
+
+  /**
+   * @var array
+   *
+   * Composer scaffold files.
+   */
+  protected $scaffoldFiles;
+
+  /**
+   * {inheritdoc}.
+   */
+  protected function configure(): void {
+    $this->setDescription('Build and push a code artifact to a Cloud Platform environment')
```

Comment:
It would be helpful to add a couple of example usages

Reply:
With only two options it's not that complex, and we already get the standard examples out of the box:
```
Usage:
  push:artifact [options] [--] [<environmentId>]
  push:artifact [<environmentAlias>]
  push:artifact myapp.dev
  push:artifact 12345-abcd1234-1111-2222-3333-0e02b2c3d470
```

But I can add another.